platform/telemetry/src/TelemetryCapability.js

```diff
@@ -300,7 +300,8 @@ define(
             var defaultRange = metadata.valuesForHints(['range'])[0];
             defaultRange = defaultRange ? defaultRange.key : undefined;
 
-            var sourceMap = _.indexBy(metadata.values(), 'key');
+            const keyBy = (array, k) => (array || []).reduce((r, x) => ({ ...r, [k ? x[k] : x]: x }), {});
```

Comment:
If we are using this function more than once in the same file, can we define it on an outer scope and reuse?

Reply:
added api/utils/array-utils, added keyBy function and imported for reuse